src/utils/sqs-huge-msg.ts

```diff
@@ -0,0 +1,158 @@
+    this.s3 = options.s3;
+    this.sqs = options.sqs;
+    this.queueName = options.queueName;
+    this.maxMessageSize = options.maxMessageSize || SqsServiceMessage.MAX_SQS_MESSAGE_SIZE;
+    this.s3Bucket = options.s3Bucket;
+    this.itemPrefix = options.itemPrefix;
+  }
+
+  private async deleteMessage(queueName: string, message: any): Promise<void> {
+    const queueUrl = await this.getQueueUrl(queueName);
+
+    if (queueUrl === undefined) {
+      return;
+    }
+
+    await this.sqs
+      .deleteMessage({
+        QueueUrl: queueUrl,
+        ReceiptHandle: message.Messages[0].ReceiptHandle,
+      }).promise();
+  }
+
+  public async getQueueUrl(queueName: string): Promise<string|undefined> {
+    const { QueueUrl } = await this.sqs
+      .getQueueUrl({
+        QueueName: queueName || this.queueName,
+      }).promise();
+
+    return QueueUrl;
+  }
+
+  public async sendMessage(queueName: string, body: string, messageAttributes?: MessageBodyAttributeMap): Promise<void|PromiseResult<AWS.SQS.SendMessageResult, AWS.AWSError>> {
+    const msgSize = Buffer.byteLength(body, 'utf-8');
+    const queueUrl = await this.getQueueUrl(queueName);
+
+    if (queueUrl === undefined) {
+      throw new Error('Queue URL not found');
+    }
+
+    if (msgSize < this.maxMessageSize) {
+      const messageConfig = {
+        QueueUrl: queueUrl,
+        MessageBody: body,
+        MessageAttributes: messageAttributes,
+      };
+
+      return this.sqs.sendMessage(messageConfig).promise();
+    }
+
+    const keyId: string = uuid();
+    const payloadId = this.itemPrefix !== undefined ? `${this.itemPrefix}/${keyId}.json` : `${keyId}.json`;
+
+    const responseBucket = await this.s3.upload({
+      Bucket: this.s3Bucket,
+      Body: body,
+      Key: payloadId,
+    }).promise();
+
+    const messageConfig = {
+      QueueUrl: queueUrl,
+      MessageBody: JSON.stringify({
+        S3Payload: {
+          Id: payloadId,
+          Key: responseBucket.Key,
+          Location: responseBucket.Location,
+        },
+      }),
+      MessageAttributes: messageAttributes,
+    };
+
+    return this.sqs.sendMessage(messageConfig).promise();
+  }
+
+  public async getMessage(queueName: string): Promise<PromiseResult<AWS.SQS.ReceiveMessageResult, AWS.AWSError>> {
+    const queueUrl = await this.getQueueUrl(queueName);
+
+    if (queueUrl === undefined) {
+      throw new Error('Queue URL not found');
+    }
+
+    const message = await this.sqs
+      .receiveMessage({ QueueUrl: queueUrl }).promise();
+
+    const messages = message.Messages;
+
+    if (!messages || messages.length === 0) {
+      throw new Error('No messages found');
+    }
+
+    const { Body } = messages[0];
+    if (Body !== undefined) {
+      messages[0].Body = await this.getMessageContent(Body);
+    }
+
+    await this.deleteMessage(queueName, message);
+
+    return message;
+  }
+
+  public async getMessageContent(body:string):Promise<string> {
```

Comment:
Return parsed body & rename

Reply:
Not required due to getMessage depending on this function and requiring a string. So this method would need wrapping in something to parse the string response and casting the output, at which point all we're doing is moving the parsing into the library and making it less flexible due to the type-casting.